charmcraft/commands/build.py

```diff
@@ -65,7 +65,7 @@ def polite_exec(cmd):
             cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
     except Exception as err:
         logger.error("Executing %s crashed with %r", cmd, err)
-        return -1
+        return 1
```

Comment:
should this be a different return code for an unknown error?
A lot of tools use a 1 vs 2

Reply:
python's doc say,

> zero is considered “successful termination” and any nonzero value is considered “abnormal termination” by shells and the like. Most systems require it to be in the range 0-127, and produce undefined results otherwise. Some systems have a convention for assigning specific meanings to specific exit codes, but these are generally underdeveloped; Unix programs generally use 2 for command line syntax errors and 1 for all other kind of errors.

But then [other] [sources] say 0-255 is fine, and 128+n is «Fatal error signal “n”». As SIGINT is 2, we *could* use 130 for that; wdyt?

[other]: https://tldp.org/LDP/abs/html/exitcodes.html
[sources]: https://shapeshed.com/unix-exit-codes/